DirectProgramming/DPC++FPGA/Tutorials/Features/read_only_cache/src/read_only_cache.cpp

```diff
@@ -0,0 +1,164 @@
+}
+
+uint16_t rand3(uint16_t &index, uint16_t &bits) {
+  bits = ((index >> 0) ^ (index >> 1) ^ (index >> 2) ^ (index >> 5)) & 1u;
+  return index = (index >> 1) | (bits << 15);
+}
+
+event runSqrtTest(sycl::queue &q, const std::vector<float> &sqrt_lut_vec,
+                 std::vector<float> &output_vec) {
+  buffer sqrt_lut_buf(sqrt_lut_vec);
+  buffer output_buf(output_vec);
+
+  event e = q.submit([&](handler &h) {
+    accessor sqrt_lut(sqrt_lut_buf, h, read_only,
+                      oa::accessor_property_list{oa::no_alias});
+    accessor output(output_buf, h, write_only,
+                    oa::accessor_property_list{oa::no_alias, no_init});
+
+    h.single_task<SqrtTest>([=]() {
+      uint16_t index = 0xFFFu;
+      uint16_t bits = 0;
+
+      for (int i = 0; i < kNumOutputs; i++)
+        output[i] = sqrt_lut[rand1(index, bits) % kLUTSize];
+
+      for (int i = 0; i < kNumOutputs; i++)
+        output[i] += sqrt_lut[rand2(index, bits) % kLUTSize];
+
+      for (int i = 0; i < kNumOutputs; i++)
+        output[i] += sqrt_lut[rand3(index, bits) % kLUTSize];
+    });
+  });
+  return e;
+}
+
+int main() {
+  // Host and kernel profiling
+  event e;
+  ulong t1_kernel, t2_kernel;
+  double time_kernel;
+
+  // Create input and output vectors
+  std::vector<float> sqrt_lut_vec(kLUTSize);
+  std::vector<float> output_vec(kNumOutputs);
+  for (int i = 0; i < kLUTSize; ++i) {
+    sqrt_lut_vec[i] = sqrt(i);
+  }
+
+// Create queue, get platform and device
+#if defined(FPGA_EMULATOR)
+  ext::intel::fpga_emulator_selector device_selector;
+#else
+  ext::intel::fpga_selector device_selector;
+#endif
+  try {
+    auto prop_list =
+        sycl::property_list{sycl::property::queue::enable_profiling()};
+
+    sycl::queue q(device_selector, dpc_common::exception_handler, prop_list);
+
+    std::cout << "\nSQRT LUT size: " << kLUTSize << "\n";
+    std::cout << "Number of outputs: " << kNumOutputs << "\n";
+
+    e = runSqrtTest(q, sqrt_lut_vec, output_vec);
+    e.wait();
+
+    // Compute kernel execution time
+    t1_kernel = e.get_profiling_info<info::event_profiling::command_start>();
+    t2_kernel = e.get_profiling_info<info::event_profiling::command_end>();
+    time_kernel = (t2_kernel - t1_kernel) / kNs;
+
+  } catch (sycl::exception const &e) {
+    // Catches exceptions in the host code
+    std::cerr << "Caught a SYCL host exception:\n" << e.what() << "\n";
+
+    // Most likely the runtime couldn't find FPGA hardware!
+    if (e.code().value() == CL_DEVICE_NOT_FOUND) {
+      std::cerr << "If you are targeting an FPGA, please ensure that your "
+                   "system has a correctly configured FPGA board.\n";
+      std::cerr << "Run sys_check in the oneAPI root directory to verify.\n";
+      std::cerr << "If you are targeting the FPGA emulator, compile with "
+                   "-DFPGA_EMULATOR.\n";
+    }
+    std::terminate();
+  }
+
+  // Compute the reference solution
+  uint16_t index = 0xFFFu;
```

Comment:
Why the magic number 0xFFFu?     A comment or better explanation would help.

Reply:
Addressed